### PR TITLE
feat!: add vertical datum in map_projector_info

### PR DESCRIPTION
**Pull request by kminoda**
## Description

In order to use `map_projector_info` in gnss_poser, we also need to define height system for whole Autoware system. Thus, I would like to add "vertical_datum" variable in `tier4_map_msgs/msg/map_projector_info` msg.

Note that **no influence on the system is intended at this moment** because the height system is not used in map_loader nor default_ad_api. This PR is for the future PR when gnss_poser will start subscribing map_projector_info.

## Related links

https://github.com/tier4/tier4_autoware_msgs/pull/98

## Tests performed

- Confirmed that logging_simulator works fine with the PR
- Build passes with the tier4_autoware_msgs PR (NOTE: **BUILD FAIL FOR THE CI IN THIS PR IS INTENDED**)
- All the test passes in `map_projection_loader`

## Notes for reviewers

Note that I will introduce constant string later on. 
This is because I will modify the supported projector_type and vertical_datum in a short period of time. If I introduce constant string now, the modifications will invoke some additional PRs. Thus, I would like to introduce constant string later when the modification on projector_type and vertical_datum are mostly fixed (probably within a month or so)

## Interface changes

- Topic change: `/map/map_projector_info` message will have an additional field in it. Also, `type` field will be renamed to `projector_type`.
- Map file change: The specification of `map_projector_info.yaml` will change accordingly.

## Effects on system behavior

Not expected

## Pre-review checklist for the PR author

The PR author **must** check the checkboxes below when creating the PR.

- [x] I've confirmed the [contribution guidelines].
- [x] The PR follows the [pull request guidelines].

## In-review checklist for the PR reviewers

The PR reviewers **must** check the checkboxes below before approval.

- [ ] The PR follows the [pull request guidelines].
- [ ] The PR has been properly tested.
- [ ] The PR has been reviewed by the code owners.

## Post-review checklist for the PR author

The PR author **must** check the checkboxes below before merging.

- [ ] There are no open discussions or they are tracked via tickets.
- [ ] The PR is ready for merge.

After all checkboxes are checked, anyone who has write access can merge the PR.

[contribution guidelines]: https://autowarefoundation.github.io/autoware-documentation/main/contributing/
[pull request guidelines]: https://autowarefoundation.github.io/autoware-documentation/main/contributing/pull-request-guidelines/

### Summary by CodeRabbit

- Refactor: Modified the `Lanelet2MapLoaderNode::on_map_projector_info` function in the `Lanelet2MapLoaderNode` class to use the `msg->projector_type` parameter instead of `msg->type` for loading the map. This may affect the behavior of the code.
- Refactor: Renamed the `type` field to `projector_type` and added a new field called `vertical_datum` in the `map_projector_info` message. Updated the `load_info_from_lanelet2_map` function, test cases, and YAML file loading functions to handle these changes.
- Refactor: Modified the `publish_kinematics` function in the `VehicleNode` class to check the `projector_type` field instead of the `type` field of the `map_projector_info_` object. The logic and functionality remain the same, but the field name has been changed.

### CODE REVIEW

File: map/map_projection_loader/test/test_node_load_mgrs_from_yaml.test.py [hunk 1]
```diff
@@ -131,7 +131,8 @@ def test_node_link(self):
         self.assertIsNotNone(
             self.received_message, "No message received on map_projector_info topic"
         )
-        self.assertEqual(self.received_message.type, yaml_data["type"])
+        self.assertEqual(self.received_message.projector_type, yaml_data["projector_type"])
+        self.assertEqual(self.received_message.vertical_datum, yaml_data["vertical_datum"])
```
[the remark this answers is not in the document]
Comment (kminoda) [reply map/map_projection_loader/test/test_node_load_mgrs_from_yaml.test.py:135]: Not applicable

File: map/map_loader/src/lanelet2_map_loader/lanelet2_map_loader_node.cpp [hunk 2]
```diff
@@ -69,8 +69,8 @@ void Lanelet2MapLoaderNode::on_map_projector_info(
   const auto center_line_resolution = get_parameter("center_line_resolution").as_double();
 
   // load map from file
-  const auto map =
-    load_map(lanelet2_filename, msg->type, msg->map_origin.latitude, msg->map_origin.longitude);
+  const auto map = load_map(
+    lanelet2_filename, msg->projector_type, msg->map_origin.latitude, msg->map_origin.longitude);
```
[the remark this answers is not in the document]
Comment (kminoda) [reply map/map_loader/src/lanelet2_map_loader/lanelet2_map_loader_node.cpp:73]: Not applicable... it's already fixed. Are you referring to the original code?

File: map/map_projection_loader/README.md [hunk 3]
```diff
@@ -44,10 +45,12 @@
...
 # map_projector_info.yaml
-type: "LocalCartesianUTM"
+projector_type: LocalCartesianUTM
+vertical_datum: WGS84
 map_origin:
-  latitude: 35.6092
-  longitude: 139.7303
+  latitude: 35.6762 # [deg]
+  longitude: 139.6503 # [deg]
+  altitude: 0.0 # [m]
```
[the remark this answers is not in the document]
Comment (kminoda) [reply map/map_projection_loader/README.md:53]: ???

File: map/map_projection_loader/src/map_projection_loader.cpp [hunk 4]
```diff
@@ -27,13 +27,15 @@
...
+  if (msg.projector_type == "MGRS") {
+    msg.vertical_datum = data["vertical_datum"].as<std::string>();
     msg.mgrs_grid = data["mgrs_grid"].as<std::string>();
-  } else if (msg.type == "LocalCartesianUTM") {
+  } else if (msg.projector_type == "LocalCartesianUTM") {
+    msg.vertical_datum = data["vertical_datum"].as<std::string>();
     msg.map_origin.latitude = data["map_origin"]["latitude"].as<double>();
     msg.map_origin.longitude = data["map_origin"]["longitude"].as<double>();
-  } else if (msg.type == "local") {
+  } else if (msg.projector_type == "local") {
```
[the remark this answers is not in the document]
Comment (kminoda) [reply map/map_projection_loader/src/map_projection_loader.cpp:38]: :cry:

**Review by RyuYamamoto — APPROVED** [review]
LGTM about map_loader

**Comment by kminoda** [comment]
@isamu-takagi Would you mind checking this PR? (as a maintainer of default_ad_api)

**Review by isamu-takagi — APPROVED** [review]
LGTM for default_ad_api

**Review by YamatoAndo — APPROVED** [review]
LGTM
please merge this  PR after merging https://github.com/tier4/tier4_autoware_msgs/pull/98

**Pull request merged**